### PR TITLE
Basic movement

### PR DESCRIPTION
**Pull request by MrJackey**
## Summary
Fixes #6.
Added basic movement to the player, enabling moving left and right.
Added jump functionality to the player.
The player currently has no friction to prevent sticking to walls.

## Visualisation
98yARNlhMt

XNB9lR4TX6

### CODE REVIEW

File: Assets/Scripts/Player/PlayerController.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			velocity.x = 0;
+
+		if (xInput != 0) {
+			float newVelocityX = velocity.x + xInput * acceleration * Time.deltaTime;
+			velocity.x = Mathf.Clamp(newVelocityX, -maxSpeed, maxSpeed);
+		} else {
+			velocity.x -= velocity.x * deacceleration * Time.deltaTime;
+		}
+
+		if (Input.GetButtonDown("Jump") && isGrounded)
```
Comment (smeas) [inline Assets/Scripts/Player/PlayerController.cs:53]: It's cheaper to check the bool before checking the input.
```suggestion
		if (isGrounded && Input.GetButtonDown("Jump"))
```

File: Assets/Scripts/Player/PlayerController.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		if (doJump) {
+			rb2D.velocity = new Vector2(velocity.x, jumpVelocity);
+			doJump = false;
+		}
+	}
+
+	private bool CheckIfGrounded() {
+		Vector3 position = transform.position;
+
+		int floorLayer = LayerMask.GetMask("Ground");
```
Comment (smeas) [inline Assets/Scripts/Player/PlayerController.cs]: You should either cache the layer mask value, or make a serialized `LayerMask` field so you can select the layer from the inspector.

File: Assets/Scripts/Player/PlayerController.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		Vector2 rayStartPosition = new Vector2(position.x, position.y - boxCollider.bounds.extents.y);
+		float rayDistance = 0.05f;
+
+		RaycastHit2D hit = Physics2D.Raycast(rayStartPosition,Vector2.down, rayDistance, floorLayer);
+
+		return hit.collider;
```
Comment (smeas) [inline Assets/Scripts/Player/PlayerController.cs]: Please use explicit null check on unity objects.
```suggestion
		return hit.collider != null;
```

**Review by smeas — CHANGES_REQUESTED** [review]

**Review by smeas — APPROVED** [review]

**Review by smeas — APPROVED** [review]

**Review by Addehman — APPROVED** [review]
Looking good to me!

**Pull request merged**